testing/asserts.ts

```diff
@@ -564,20 +573,23 @@ export function assertObjectMatch(
           filtered[key] = value;
           continue;
         } // On nested objects references, build a filtered object recursively
-        else if (typeof value === "object") {
+        else if (typeof value === "object" && value !== null) {
```

Comment:
Real change is checking that the value is not null so it doesn't get sent for another recursion